config/routes.rb

```diff
@@ -1,10 +1,16 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get "users/:id" => "users#show", as: :user
+  get "users/:user_id/posts" => "users/posts#index", as: :user_posts
+  get "users/:user_id/posts/:id" => "users/posts#show", as: :user_post
+
+
+
+  resources :users, only: [:index, :show] do
+  resources :posts, only: [:index, :show], module: :users
+  end
```

Comment:
- [ ] Same here, you created some routes twice, Please edit your routes to be like the suggestion below.
```suggestion
  resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
  end
```